cdp_scrapers/instances/houston.py

```diff
@@ -0,0 +1,283 @@
+            All elements in the page that we want to scrape
+
+        Returns:
+        ----------------
+        List[ingestion_models.EventMinutesItem]
+            All the event minute items gathered from the event on the page
+        """
+        log.info("start get items")
+        event_minutes_items = []
+        all_items = self.remove_extra_type(event).find_all("td", {"class": "style4"})
+        for item in all_items:
+            name = ""
+
+            for i in item.stripped_strings:
+                name = name + " " + repr(i).replace("'", "")
+
+            if name is not None and name != "":
+                event_minutes_items.append(
+                    ingestion_models.EventMinutesItem(
+                        minutes_item=ingestion_models.MinutesItem(name.strip())
+                    )
+                )
+
+        return event_minutes_items
+
+    # Big Functions
+    def get_diff_yearid(self, time: datetime) -> str:
+        """
+        Get the events in different years as the events for different
+        years are stored in different tabs. Can get multiple events
+        across years.
+
+        Parameters:
+        ---------------
+        time: datetime
+            The date of the event we are trying to parse
+
+        Returns:
+        ---------------
+        str
+            The year id that can locate the year tab where the event is stored
+        """
+        year = str(time.year)
+        year_id = "city-council-" + year
+        return year_id
+
+    def get_date_mainlink(self, element: BeautifulSoup) -> str:
+        """
+        Find the main link for one event.
+
+        Parameters:
+        --------------
+        time: datetime
+            The date of one event
+
+        Returns:
+        --------------
+        str
+            The main link, make agenda and video url in other function
+        """
+        link_post = element.find("a")["href"]
+        link = f"https://houstontx.new.swagit.com/{link_post}"
+        return link
+
+    def get_agenda(self, element: BeautifulSoup) -> Union[Tag, NavigableString, None]:
+        """
+        Get event agenda for a specific details page
+
+        Parameters:
+        ----------------
+        event_time: datetime
+            The date we want to get agenda
+
+        Returns:
+        ----------------
+        Tag
+            The agenda web page we want parse
+        """
+        link = self.get_date_mainlink(element)
+        agenda_link = link + "/agenda"
+        page = requests.get(agenda_link)
+        event = BeautifulSoup(page.content, "html.parser")
+        form1 = event.find("form", id="Form1")
+        return form1
+
+    def get_event(self, element_list) -> ingestion_models.EventIngestionModel:
+        """
+        Parse one event at a specific date. City council meeting information for
+        a specific date
+
+        Parameters:
+        --------------
+        event_time: datetime
+            Meeting date
```

Comment:
The function signature uses `element_list` as a parameter, but the docstring uses `event_time`. Also, typing is missing for the parameter in the function signature.

Reply:
Done

Reply:
One more nitpick Shak 😅 currently we have `element_list: []`. Personally not too thrilled about that. `[]` is an object, not a type. i.e. To me, it would be better to do something like `element_list: List`.

Actually, in fact, that's not the correct type at all. If you look at [how `get_event()` is called](https://github.com/CouncilDataProject/cdp-scrapers/blob/2b5b3700c6082ca8c1c3ed49bc85931ed5a2af0b/cdp_scrapers/instances/houston.py#L248-L249), it's called with a _single_ "thing" as the parameter.

```Python
for element in elements:
    events.append(self.get_event(element))
```

Reply:
Well, I guess `element` there _could_ be a list of some kind. But if so, then that is very misleading. So, either just change the typing in the function signature from `[]` to something like `List` (nitpicky LOL); or, change the variable name at the call site (in `get_events()`) so that it's clear it's a list (`element` just sounds like "one thing").

Reply:
Good catch! I think I had a mixing of previous names and current implementation.

In this function `get_all_elements_in_range`, I need to return both the date and the element, so I am using a dictionary.